butterfly-utilities/src/main/java/com/paypal/butterfly/utilities/operations/pom/PomRemoveDependencyExclusion.java

```diff
@@ -0,0 +1,165 @@
+        ifNotPresent = ChangeOrRemoveElement.IfNotPresent.Warn;
+        return this;
+    }
+
+    @Override
+    public PomRemoveDependencyExclusion noOpIfNotPresent() {
+        ifNotPresent = ChangeOrRemoveElement.IfNotPresent.NoOp;
+        return this;
+    }
+
+    public Exclusion getExclusion() {
+        return this.exclusion;
+    }
+
+    @Override
+    public String getDescription() {
+        return String.format(DESCRIPTION, exclusion.getGroupId(), exclusion.getArtifactId(), groupId,
+                artifactId, getRelativePath());
+    }
+
+    @Override
+    protected TOExecutionResult pomExecution(String relativePomFile, Model model) {
+        TOExecutionResult result = null;
+        String details;
+
+        Dependency dependency = getDependency(model, groupId, artifactId);
+        if(dependency == null) {
+            dependency = getDependencyFromDependencyManagement(model);
+        }
+        if (dependency != null) {
+
+            List<Exclusion> exclusions = dependency.getExclusions();
+
+            if(exclusions != null && exclusions.size() > 0) {
+                removeExclusionFromDependency(dependency, exclusions);
+                details = String.format("Exclusion %s:%s has been removed from dependency %s:%s in POM file %s",
+                        exclusion.getGroupId(), exclusion.getArtifactId(), groupId, artifactId, relativePomFile);
+                result = TOExecutionResult.success(this, details);
+            } else {
+                details = String.format("Exclusion %s:%s not present in dependency %s:%s in POM file %s",
+                        exclusion.getGroupId(), exclusion.getArtifactId(), groupId, artifactId, relativePomFile);
+                result = getTOExecutionResult(details);
+            }
+        } else {
+            details = String.format("Exclusion %s:%s has not been removed as dependency %s:%s in POM file %s because dependency is not present",
+                    exclusion.getGroupId(), exclusion.getArtifactId(), groupId, artifactId, relativePomFile);
+            result = getTOExecutionResult(details);
+        }
+        return result;
+    }
+
+    private TOExecutionResult getTOExecutionResult(String details) {
+        TOExecutionResult result;
+        switch (ifNotPresent) {
+            case Warn:
+                result = TOExecutionResult.warning(this,
+                        new TransformationOperationException(details));
+                break;
+            case NoOp:
+                result = TOExecutionResult.noOp(this, details);
+                break;
+            case Fail:
+                // Fail is the default
+            default:
+                result = TOExecutionResult.error(this,
+                        new TransformationOperationException(details));
+                break;
+        }
+        return result;
+    }
+
+    private Dependency getDependencyFromDependencyManagement(Model model) {
```

Comment:
Why you didn't use `getManagedDependency` from super class?

Reply:
Thanks for mentioning that @fabiocarvalho777 
I have refactored to use getManagedDependency